profiles/tests/test_gql_federation.py

```diff
@@ -3,54 +3,71 @@
 from guardian.shortcuts import assign_perm
 
 from open_city_profile.tests.asserts import assert_match_error_code
-from profiles.schema import ProfileNode
-from profiles.tests.factories import ProfileFactory
+from profiles.schema import AddressNode, ProfileNode
+from profiles.tests.factories import AddressFactory, ProfileFactory
 from services.tests.factories import ServiceConnectionFactory
 
+GRAPHQL_SDL_QUERY = """
+    query {
+        _service {
+            sdl
+        }
+    }
+"""
 
-def test_profile_node_exposes_key_for_federation_gateway(anon_user_gql_client):
-    query = """
-        query {
-            _service {
-                sdl
+
+PROFILE_ENTITY_QUERY = """
+    query ($_representations: [_Any!]!) {
+        _entities(representations: $_representations) {
+            ... on ProfileNode {
+                id
+                firstName
             }
         }
-    """
-
-    executed = anon_user_gql_client.execute(query)
-    assert (
-        'type ProfileNode implements Node  @key(fields: "id")'
-        in executed["data"]["_service"]["sdl"]
-    )
+    }
+"""
 
 
-def test_profile_connection_schema_matches_federated_schema(anon_user_gql_client):
-    query = """
-        query {
-            _service {
-                sdl
+ADDRESS_ENTITY_QUERY = """
+    query ($_representations: [_Any!]!) {
+        _entities(representations: $_representations) {
+            ... on AddressNode {
+                id
+                address
+                postalCode
             }
         }
-    """
+    }
+"""
```

Comment:
These can be merged into one query now:

```python
ENTITY_QUERY = """
    query ($_representations: [_Any!]!) {
        _entities(representations: $_representations) {
            ... on ProfileNode {
                id
                firstName
            }
            ... on AddressNode {
                id
                address
                postalCode
            }
        }
    }
"""
```

Reply:
Ok, Github doesn't present this very clearly. The point was, that `PROFILE_ENTITY_QUERY` and `ADDRESS_ENTITY_QUERY` can be replaced with a single `ENTITY_QUERY` that handles both kinds of entities.

Reply:
Makes sense. Done in 53981a6 👍🏼